sherpa-onnx/csrc/provider-config.cc

```diff
@@ -0,0 +1,158 @@
+        trt_max_partition_iterations);
+    return false;
+  }
+  if (trt_min_subgraph_size < 0) {
+    SHERPA_ONNX_LOGE("trt_min_subgraph_size: '%u' is not valid.",
+        trt_min_subgraph_size);
+    return false;
+  }
+  if (trt_fp16_enable != true || trt_fp16_enable != false) {
+    SHERPA_ONNX_LOGE("trt_fp16_enable: '%d' is not valid.", trt_fp16_enable);
+    return false;
+  }
+  if (trt_detailed_build_log != true || trt_detailed_build_log != false) {
+    SHERPA_ONNX_LOGE("trt_detailed_build_log: '%d' is not valid.",
+        trt_detailed_build_log);
+    return false;
+  }
+  if (trt_engine_cache_enable != true || trt_engine_cache_enable != false) {
+    SHERPA_ONNX_LOGE("trt_engine_cache_enable: '%d' is not valid.",
+        trt_engine_cache_enable);
+    return false;
+  }
+  if (trt_timing_cache_enable != true || trt_timing_cache_enable != false) {
+    SHERPA_ONNX_LOGE("trt_timing_cache_enable: '%d' is not valid.",
+        trt_timing_cache_enable);
+    return false;
+  }
+  if (trt_dump_subgraphs != true || trt_dump_subgraphs != false) {
+    SHERPA_ONNX_LOGE("trt_dump_subgraphs: '%d' is not valid.",
+        trt_dump_subgraphs);
+    return false;
+  }
+
+  return true;
+}
+
+std::string TensorrtConfig::ToString() const {
+  std::ostringstream os;
+
+  os << "TensorrtConfig(";
+  os << "trt_max_workspace_size=\"" << trt_max_workspace_size << "\", ";
+  os << "trt_max_partition_iterations=\""
+      << trt_max_partition_iterations << "\", ";
+  os << "trt_min_subgraph_size=\"" << trt_min_subgraph_size << "\", ";
+  os << "trt_fp16_enable=\""
+      << (trt_fp16_enable? "True" : "False") << "\", ";
+  os << "trt_detailed_build_log=\""
+      << (trt_detailed_build_log? "True" : "False") << "\", ";
+  os << "trt_engine_cache_enable=\""
+      << (trt_engine_cache_enable? "True" : "False") << "\", ";
+  os << "trt_engine_cache_path=\""
+      << trt_engine_cache_path.c_str() << "\", ";
+  os << "trt_timing_cache_enable=\""
+      << (trt_timing_cache_enable? "True" : "False") << "\", ";
+  os << "trt_timing_cache_path=\""
+      << trt_timing_cache_path.c_str() << "\",";
+  os << "trt_dump_subgraphs=\""
+      << (trt_dump_subgraphs? "True" : "False") << "\" )";
+  return os.str();
+}
+
+void ProviderConfig::Register(ParseOptions *po) {
+  cuda_config.Register(po);
+  trt_config.Register(po);
+
+  po->Register("device", &device, "GPU device index for CUDA and Trt EP");
+  po->Register("provider", &provider,
+               "Specify a provider to use: cpu, cuda, coreml");
+}
+
+bool ProviderConfig::Validate() const {
+  if (device < 0) {
+    SHERPA_ONNX_LOGE("device: '%d' is invalid.", device);
+    return false;
+  }
+  return true;
+}
+
+std::string ProviderConfig::ToString() const {
+  std::ostringstream os;
+
+  os << "ProviderConfig(";
+  os << "device=\"" << device << "\", ";
+  os << "provider=\"" << provider << "\", ";
+  os << "cuda_config=\"" << cuda_config.ToString() << "\", ";
```

Comment:
```suggestion
  os << "cuda_config=" << cuda_config.ToString() << ", ";
```

Note that `""` is needed only for string variables. Please fix similar issues in your PR.